numpyro/util.py

```diff
@@ -149,11 +148,9 @@ def _body_fn(i, vals):
                 val = body_fun(val)
                 collection.append(jit(ravel_fn)(val))
                 if diagnostics_fn:
-                    # TODO: set refresh=True when its performance issue is resolved
                     t.set_postfix_str(diagnostics_fn(val), refresh=False)
 
-        # XXX: jax.numpy.stack/concatenate is currently so slow
-        collection = onp.stack(collection)
+        collection = np.stack(collection)
```

Comment:
+1. Is the perf disparity completely resolved?

Reply:
It is not completely resolved. It is still slow for large num_samples, but not so slow as before. So I'll revert this change. :)